File: Day 7/2 Sliding Window/solution.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        
        int MaxProfit = 0;
        
        int buyTime = 0;
        
        for (int sellTime=1; sellTime< prices.Length; sellTime++){
            
            //
            if(prices[buyTime] > prices[sellTime]){
                buyTime = sellTime;
            }
            
            MaxProfit = Math.Max(prices[sellTime] - prices[buyTime], MaxProfit);
            
        }
        
        return MaxProfit;
    }
}

File: Day 7/2 Sliding Window/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        buy_time = 0
        for sell_time in range(1, len(prices)):
            if prices[buy_time] > prices[sell_time]:
                buy_time = sell_time
            max_profit = max(max_profit, prices[sell_time] - prices[buy_time])
        return max_profit
